### ISSUE TITLE
no-dupe-args and no-redeclare false positives in 20.1.0

### ISSUE THREAD

**Issue by vkrol** [eslint/typescript-eslint-parser#535]
**What version of TypeScript are you using?**
3.1.4

**What version of `typescript-eslint-parser` are you using?**
20.1.0

**What code were you trying to parse?**

```ts
function foo({ bar }: { bar: string }) {
	console.log(bar);
}
```

**What did you expect to happen?**
No errors.

**What happened?**
```
  1:1   error  Duplicate param 'bar'         no-dupe-args
  1:25  error  'bar' is already defined      no-redeclare
```

**Comment by chanlito** [comment]
I'm experiencing this issue as well after updating my vue-cli ts deps to latest.

**Comment by mysticatea** [comment]
Thank you for this report.

I wonder we can disable both rules `no-dupe-args` and `no-redeclare` because TypeScript verifies those itself?

**Comment by chanlito** [comment]
That's my solution right now. Turn them off.

**Comment by milesj** [comment]
This is happening for a ton of stuff, including `no-shadow`. Here's some output from my lints.

```
  3:90  error  'Theme' is already declared in the upper scope  no-shadow

  20:39  error  'Options' is already declared in the upper scope  no-shadow

  28:3   error  Duplicate param 'id'          no-dupe-args
  28:3   error  Duplicate param 'name'        no-dupe-args
  28:3   error  Duplicate param 'tagName'     no-dupe-args
  28:3   error  Duplicate param 'type'        no-dupe-args
  33:8   error  'id' is already defined       no-redeclare
  33:21  error  'name' is already defined     no-redeclare
  33:36  error  'tagName' is already defined  no-redeclare
  33:54  error  'type' is already defined     no-redeclare
```

Is this related to the eslint visitor keys change?

**Comment by mysticatea** [comment]
> Is this related to the eslint visitor keys change?

Yes. As I mentioned on https://github.com/eslint/typescript-eslint-parser/pull/516#issuecomment-428362948, the change changed traversal in scope analysis, too. (I had assumed people has disabled rules of variables because `tsc` verifies it)

I will try to fix the scope analysis soon.

**Comment by ffxsam** [comment]
Just stumbled upon this too, would be great to have it ignore TS syntax.

**Comment by vkrol** [comment]
@mystictea thank you!

**Comment by vkrol** [comment]
@mysticatea thank you!
